server/backend/check.ml

```diff
@@ -718,7 +723,8 @@ let run ~debug ~cap_file ~on_finished ~conf oca_cache workdir =
         let* cap = get_cap ~stderr ~cap_file in
         let* opam_repo, opam_repo_commit = get_commit_hash_default conf in
         let* extra_repos = get_commit_hash_extra_repos conf in
-        let* cache = cache ~stderr ~conf in
+        let with_dune = List.exists Intf.Switch.with_dune switches in
```

Comment:
I'm not sure to understand this part? Don't we want only the switches we are going to use to be `dune` compatible?

Reply:
It is really only about the warning that `Check.cache` prints: if *any* of the switches uses `dune` and `enable_dune_cache` is disabled, display the warning or be silent otherwise.

The issue is that `cache` does not take the list of switches (since the cache is independent of them, it is not per-switch but rather per-backend-server-instance), thus I either need to pass it the switches or let it know some other way whether this warning should be printed.

(Fun fact, making that error message work required adding more arguments to functions than making the `with-dune` feature work per-switch)